src/plugins/intel_cpu/src/nodes/multinomial.cpp

```diff
@@ -0,0 +1,287 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "multinomial.hpp"
+
+#include "ie_ngraph_utils.hpp"
+#include "openvino/op/multinomial.hpp"
+#include "utils/bfloat16.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Multinomial::Multinomial(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op, PortMask(NUM_SAMPLES_PORT))) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_CPU_NODE_ERR(errorMessage);
+    }
+
+    auto multinomial_op = as_type_ptr<op::v13::Multinomial>(op);
+    m_with_replacement = multinomial_op->get_with_replacement();
+    m_global_seed = multinomial_op->get_global_seed();
+    m_log_probs = multinomial_op->get_log_probs();
+    m_op_seed = multinomial_op->get_op_seed();
+
+    m_num_samples_precision = ov::element::i32;
+    m_output_precision = multinomial_op->get_convert_type();
+
+    constant = ConstantType::NoConst;
+
+    m_const_batch = op->get_input_partial_shape(PROBS_PORT)[0].is_static();
+    m_const_inputs[PROBS_PORT] = is_type<op::v0::Constant>(op->get_input_node_ptr(PROBS_PORT));
+    m_const_inputs[NUM_SAMPLES_PORT] = is_type<op::v0::Constant>(op->get_input_node_ptr(NUM_SAMPLES_PORT));
+}
+
+bool Multinomial::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v13::Multinomial::get_type_info_static()) {
+            errorMessage = "Only Multinomial operation from the opset13 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void Multinomial::getSupportedDescriptors() {
+    if (getParentEdges().size() != 2) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().size() != 1) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void Multinomial::initSupportedPrimitiveDescriptors() {
+    m_probs_precision = getOriginalInputPrecisionAtPort(PROBS_PORT);
+    if (!one_of(m_probs_precision, ov::element::f32, ov::element::f16, ov::element::bf16)) {
+        m_probs_precision = ov::element::f32;
+    }
+
+    addSupportedPrimDesc({{LayoutType::ncsp, m_probs_precision, m_const_inputs[PROBS_PORT]},
+                          {LayoutType::ncsp, m_num_samples_precision, m_const_inputs[NUM_SAMPLES_PORT]}},
+                         {{LayoutType::ncsp, m_output_precision}},
+                         ref_any);
+}
+
+std::string Multinomial::getPrimitiveDescriptorType() const {
```

Comment:
Why we need to override that function at all? Should work well w/o it.

Reply:
Thanks for the suggestion, fixed in [this followup PR](https://github.com/openvinotoolkit/openvino/pull/21469)